classes/submission/PKPAuthorDAO.inc.php

```diff
@@ -297,25 +298,23 @@ function deleteObject($author) {
 
 	/**
 	 * Delete an author by ID.
-	 * @param $authorId int
-	 * @param $submissionId int optional
+	 * @param $authorId int Author ID
+	 * @param $submissionId int Optional submission ID.
 	 */
 	function deleteById($authorId, $submissionId = null) {
 		$params = array((int) $authorId);
 		if ($submissionId) $params[] = (int) $submissionId;
-		$returner = $this->update(
```

Comment:
There's a common mistaken assumption in many of the DAO functions that the update function will return the number of affected rows. It doesn't; that's what this logic was assuming, so I removed it.